lib/puppet/provider/ios_ntp_config/cisco_ios.rb

```diff
@@ -0,0 +1,79 @@
+require_relative '../../../puppet_x/puppetlabs/cisco_ios/check'
+require_relative '../../util/network_device/cisco_ios/device'
+require_relative '../../../puppet_x/puppetlabs/cisco_ios/utility'
+
+# Register legacy Puppet provider instance for compatibility with other netdev_stdlib providers
+# Please do not do this with other Resource API based providers
+Puppet::Type.type(:ntp_config).provide(:ios) do
+end
+
+# NTP Config Puppet Provider for Cisco IOS devices
+class Puppet::Provider::IosNtpConfig::CiscoIos
+  def self.commands_hash
+    local_commands_hash = PuppetX::CiscoIOS::Utility.load_yaml(File.expand_path(__dir__) + '/command.yaml')
+    ntp_config_commands_hash = PuppetX::CiscoIOS::Utility.load_yaml(File.expand_path(__dir__) + '/../ntp_config/command.yaml')
+    @commands_hash = local_commands_hash.merge(ntp_config_commands_hash) { |_key, oldval, newval| (oldval.to_a + newval.to_a).to_h }
+  end
+
+  def self.instances_from_cli(output)
+    new_instance_fields = []
+    new_instance = PuppetX::CiscoIOS::Utility.parse_resource(output, commands_hash)
+    new_instance[:name] = 'default'
+    new_instance[:authenticate] = !new_instance[:authenticate].nil?
+    new_instance[:update_calendar] = (new_instance[:update_calendar]) ? true : false
+    unless new_instance[:trusted_key].nil?
```

Comment:
This provider should only really need to manage the parsing of `update_calendar`, the rest of the information can come from the base `NtpConfig` provider, something similar to: https://github.com/puppetlabs/cisco_ios/blob/master/lib/puppet/provider/ios_radius_global/cisco_ios.rb#L16-L17 

Where `attributes` is the new field of `ios_radius_global` - but all the existing values come from `radius_global`